scipy/signal/signaltools.py

```diff
@@ -2324,9 +2324,57 @@ def invres(r, p, k, tol=1e-3, rtype='avg'):
     return b, a
 
 
+def _compute_factors(roots, multiplicity):
+    """Compute the total polynomial divided by factors for each root."""
+    suffixes = []
+    current = np.array([1])
```

Comment:
Maybe you need `np.array([1], roots.dtype)` or something like this? Might fix the error, not sure